src/Compilers/CSharp/Test/Semantic/Semantics/InitOnlyMemberTests.cs

```diff
@@ -0,0 +1,2255 @@
+                //     string I2.Property { init { } } // 4
+                Diagnostic(ErrorCode.ERR_ExplicitPropertyMismatchInitOnly, "init").WithArguments("IWithInitWithExplicitImplementationOfI2.I2.Property.set", "I2.Property.set").WithLocation(40, 26),
+                // (45,26): error CS8805: Accessors 'IWithoutInitWithExplicitImplementationOfI1.I1.Property.set' and 'I1.Property.set' should both be init-only or neither
+                //     string I1.Property { set { } } // 5
+                Diagnostic(ErrorCode.ERR_ExplicitPropertyMismatchInitOnly, "set").WithArguments("IWithoutInitWithExplicitImplementationOfI1.I1.Property.set", "I1.Property.set").WithLocation(45, 26),
+                // (62,52): error CS8804: 'CWithImplementationWithInitOnly' does not implement interface member 'I2.Property.set'. 'CWithImplementationWithInitOnly.Property.set' cannot implement 'I2.Property.set' because it does not match by init-only.
+                // public class CWithImplementationWithInitOnly : I1, I2 // 6
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberWrongInitOnly, "I2").WithArguments("CWithImplementationWithInitOnly", "I2.Property.set", "CWithImplementationWithInitOnly.Property.set").WithLocation(62, 52),
+                // (66,51): error CS8804: 'CWithImplementationWithoutInitOnly' does not implement interface member 'I1.Property.set'. 'CWithImplementationWithoutInitOnly.Property.set' cannot implement 'I1.Property.set' because it does not match by init-only.
+                // public class CWithImplementationWithoutInitOnly : I1, I2 // 7
+                Diagnostic(ErrorCode.ERR_CloseUnimplementedInterfaceMemberWrongInitOnly, "I1").WithArguments("CWithImplementationWithoutInitOnly", "I1.Property.set", "CWithImplementationWithoutInitOnly.Property.set").WithLocation(66, 51)
+                );
+        }
+
+        [Fact]
+        public void EventWithInitOnly()
+        {
+            string source = @"
+public class C
+{
+    public event System.Action Event
+    {
+        init { }
+    }
+}
+";
+            var comp = CreateCompilation(new[] { source, IsExternalInitTypeDefinition }, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics(
+                // (4,32): error CS0065: 'C.Event': event property must have both add and remove accessors
+                //     public event System.Action Event
+                Diagnostic(ErrorCode.ERR_EventNeedsBothAccessors, "Event").WithArguments("C.Event").WithLocation(4, 32),
+                // (6,9): error CS1055: An add or remove accessor expected
+                //         init { }
+                Diagnostic(ErrorCode.ERR_AddOrRemoveExpected, "init").WithLocation(6, 9)
+                );
+        }
+
+        [Fact]
+        public void IndexerWithInitOnly()
+        {
+            string source = @"
+public class C
+{
+    public string this[int i]
+    {
+        init { }
```

Comment:
>init { } [](start = 8, length = 8)

The speclet does not mention indexers, but LDM notes [https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-13.md#decision-3](https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-13.md#decision-3) indicate we should support them. Please make sure to adjust the spec.

Reply:
Done

---
In reply to: [415207408](https://github.com/dotnet/roslyn/pull/43275#discussion_r415207408) [](ancestors = 415207408)